### PR TITLE
Allow the library to work with a recent version of Lumen

### PR DESCRIPTION
**Pull request by ls-paul-julien-vauthier**
What, a new repo? 😄 

This will allow us to trace individuals requests in the logs. I managed to test this locally using ProvisioningService as an example.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+version: 2
+
+jobs:
+  build:
+    docker:
+      - image: circleci/php:7.3-node-browsers
```
Comment (miqdadamirali) [inline .circleci/config.yml:6]: don't wanna do 7.4?

Comment (ls-paul-julien-vauthier) [reply .circleci/config.yml:6]: I made it PHP 7.3 since it's lib that may be used by non PHP 7.4 apps.

**Comment by ls-paul-julien-vauthier** [comment]
The PR that goes with it (waiting for this PR to be merged, then I'll change the `composer.json`.

File: composer.json [hunk 2]
```diff
@@ -1,9 +1,9 @@
 {
-    "name": "bonsi/laravel-tracetag",
-    "description": "Trace a request through your application using a tag",
+    "name": "lightspeed/laravel-tracetag",
+    "description": "Trace a request through your application using a tag. Forked from bonsi/laravel-tracetag",
     "type": "library",
     "require": {
-        "php": ">=7.0",
+        "php": ">=7.1",
```
Comment (miqdadamirali) [inline composer.json:6]: 7.3?

File: src/TraceTagServiceProvider.php [hunk 3]
```diff
@@ -29,20 +28,38 @@ public function boot()
         $this->publishes([$source => config_path('trace-tag.php')]);
         $this->mergeConfigFrom($source, 'trace-tag');
 
-        if($this->app->config->get('trace-tag.middleware.enabled', false))
+        if(config('trace-tag.middleware.enabled', false))
```
Comment (miqdadamirali) [inline src/TraceTagServiceProvider.php]:
```suggestion
        if (config('trace-tag.middleware.enabled', false))
```

File: src/TraceTagServiceProvider.php [hunk 4]
```diff
@@ -29,20 +28,38 @@ public function boot()
...
         {
-            $this->registerMiddleware(TraceTagMiddleware::class);
+            if($this->isLumenApp())
```
Comment (miqdadamirali) [inline src/TraceTagServiceProvider.php]:
```suggestion
            if ($this->isLumenApp())
```

File: src/TraceTagServiceProvider.php [hunk 5]
```diff
@@ -29,20 +28,38 @@ public function boot()
...
+            {
+                Log::warning(sprintf('%s middleware needs to be registered in app.php for an Lumen app'));
```
Comment (miqdadamirali) [inline src/TraceTagServiceProvider.php]: looks like it's missing param for `%s`

File: src/TraceTagServiceProvider.php [hunk 6]
```diff
@@ -29,20 +28,38 @@ public function boot()
...
+                $kernel = $this->app[Kernel::class];
+                $kernel->pushMiddleware(TraceTagMiddleware::class);
+            }
         }
     }
 
+    /**
+     * Check against the app version to determine whether or not the app is Laravel or Lumen.
+     *
+     * @return bool
```
Comment (miqdadamirali) [inline src/TraceTagServiceProvider.php]:
```suggestion
```
redundant given the return type in function?

**Review by miqdadamirali — APPROVED** [review]
A few comments, otherwise looking good

**Pull request merged**